SU2_CFD/src/solvers/CGradientSmoothingSolver.cpp

```diff
@@ -909,28 +912,31 @@ void CGradientSmoothingSolver::ReadSensFromGeometry(const CGeometry* geometry) {
   }
 }
 
-void CGradientSmoothingSolver::WriteSensitivity(CGeometry* geometry, const CConfig* config, unsigned long val_marker) {
+void CGradientSmoothingSolver::WriteSensitivity(CGeometry* geometry, const CConfig* config) {
   unsigned long iPoint, total_index;
   unsigned int iDim;
   su2double* normal;
   su2double norm;
 
-  /*--- split between surface and volume first, to avoid mpi ranks with no part of the marker to write back nonphysical solutions in the surface case ---*/
+  /*--- Split between surface and volume first. ---*/
   if ( config->GetSmoothOnSurface() ) {
-    if( val_marker!=NOT_AVAILABLE ) {
-      for (unsigned long iVertex =0; iVertex<geometry->nVertex[val_marker]; iVertex++)  {
-        iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
-        normal = geometry->vertex[val_marker][iVertex]->GetNormal();
-        norm = 0.0;
-        for (iDim = 0; iDim < nDim; iDim++) {
-          norm += normal[iDim]*normal[iDim];
-        }
-        norm = sqrt(norm);
-        for (iDim = 0; iDim < nDim; iDim++) {
-          normal[iDim] = normal[iDim] / norm;
-        }
-        for (iDim = 0; iDim < nDim; iDim++) {
-          nodes->SetSensitivity(iPoint, iDim, normal[iDim]*LinSysSol[iPoint]);
+    /*--- Write back values for all design boundaries in the surface case. ---*/
+    for (unsigned short iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++) {
+      if (config->GetMarker_All_DV(iMarker) == YES) {
+        for (unsigned long iVertex = 0; iVertex < geometry->nVertex[iMarker]; iVertex++) {
+          iPoint = geometry->vertex[iMarker][iVertex]->GetNode();
```

Comment:
This type of local, no need to go over the entire PR, just keep it in mind.
```suggestion
          const auto iPoint = geometry->vertex[iMarker][iVertex]->GetNode();
```

Reply:
Thanks for clarifying, I will keep it in mind for the future.